### ISSUE TITLE
support fluentd v0.14

### ISSUE THREAD

**Issue by pitr** [uken/fluent-plugin-elasticsearch#176]
first discovered in (unrelated) #175, there's a few issues:

- [x] `TypeError: no implicit conversion of nil into String` in `eval` on out_elasticsearch_dynamic.rb:130
- [x] in `test_configure`, `instance.port` is an int, not string
- [x] in `test_legacy_hosts_list`, `host3[:port]` is 0
- [x] failing `test_writes_to_speficied_port`

**Comment by eduardolundgren** [comment]
Any update on fluentd v0.14 support? Thank you.

**Comment by pitr** [comment]
Nope. PR's welcomed

**Comment by eduardolundgren** [comment]
@pragmaticivan Could you try looking if we can add support for v0.14?

**Comment by pitr** [comment]
released with v1.6.0

**Comment by cosmo0920** [comment]
Hi, this plugin will support *really* Fluentd v0.14?
Because current implementation depends on `Fluent::Compat::BufferredOutput` (= `Fluent::BufferedOutput`) and compatible layer.
Using features which are newly added in v0.14 needs to inherits `Fluent::Plugin::Output` class in this plugin.

**Comment by pitr** [comment]
seems like it's not warning about deprecation yet, so for now we are ok. However, a PR that moves plugin to `Fluent::Plugin::Output` would be greatly appreciated

**Comment by pitr** [comment]
Additionally,
1. None of the other [top plugins](http://www.fluentd.org/plugins/all#input-output) converted to `Fluent::Plugin::Output`
2. [Plugin Development Docs](http://docs.fluentd.org/articles/plugin-development) still uses `Fluent::Output` and friends.

**Comment by pragmaticivan** [comment]
Please check http://www.fluentd.org/blog/fluentd-v0.14.0-has-been-released

```
Sub-second time
In Fluentd v0.12 or earlier, time was represented as seconds from Epoch. Events generated in a second had same time. Currently, Fluentd v0.14 generates time in nanosecond resolution. 3rd party plugins which uses Fluent::Engine.now also generates these new time objects without any updates in v0.14.

This new time object behaves as Integer for compatibility with v0.12. If you want to get sub-second data from these time, update your plugins to call time.nsec to get sub-second values (and write it to anywhere you want).

This feature is implemented only on Fluentd v0.14, and Fluentd v0.12 will crash if it receives binary data including nanosecond resolution time forwarded from Fluentd v0.14. So users should be sure to specify time_as_integer true in configuration for forward output plugin if you want to transfer data from Fluentd v0.14 to Fluent v0.12 (It is true in default at v0.14.0, but will be turned into false in default at any point of future).
```

for some reason milliseconds still not returning in this version. Maybe because of this feature.
